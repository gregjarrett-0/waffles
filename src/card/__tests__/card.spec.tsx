import { render, fireEvent } from '@testing-library/react';

import { Card } from '../index';

describe('Card', () => {
  it('renders as a section by default', () => {
    const { container } = render(
      <Card>
        <p>Taylor Swift</p>
      </Card>,
    );

    const card = container.querySelector('section');

    expect(card).toBeInTheDocument();
  });

  it('renders as custom element provided via as prop', () => {
    const { container } = render(
      <Card as="div">
        <p>Taylor Swift</p>
      </Card>,
    );

    const card = container.querySelector('div');

    expect(card).toBeInTheDocument();
  });

  it('sets the data attribute on the card', () => {
    const { getByTestId } = render(
      <Card data-testid="taylors-card">
        <p>Taylor Swift</p>
      </Card>,
    );

    const button = getByTestId('taylors-card');

    expect(button).toBeInTheDocument();
  });

  it('renders snapshot of content', () => {
    const { container } = render(
      <Card>
        <h3>Taylor Swift</h3>
        <p>Better than Jimmy Page.</p>
      </Card>,
    );

    const card = container.querySelector('section');

    expect(card).toMatchSnapshot();
  });

  it('renders snapshot of the card with hover disabled', () => {
    const { container } = render(
      <Card disableHoverEffect>
        <h3>Taylor Swift</h3>
        <p>Better than Jimmy Page.</p>
      </Card>,
    );

    const card = container.querySelector('section');

    expect(card).toMatchSnapshot();
  });

  it('renders snapshot of headstones and content', () => {
    const { container } = render(
      <Card
        headstone={
          <>
            <Card.HeadstoneItem content="A" />
            <Card.HeadstoneItem content="B" />
          </>
        }
      >
        <p>Taylor Swift</p>
      </Card>,
    );

    const card = container.querySelector('section');

    expect(card).toMatchSnapshot();
  });

  it('renders snapshot of focused card', () => {
    const { getByText } = render(
      <Card as="a" href="http://taylorswift.com">
        Taylor Swift Website
      </Card>,
    );

    const card = getByText('Taylor Swift Website');
    fireEvent.focus(card);

    expect(card).toMatchSnapshot();
  });

  it('renders snapshot of inverted', () => {
    const { container } = render(
      <Card inverted headstone={<Card.HeadstoneItem content="A" />}>
        <p>Taylor Swift</p>
      </Card>,
    );

    const card = container.querySelector('section');

    expect(card).toMatchSnapshot();
  });
});
