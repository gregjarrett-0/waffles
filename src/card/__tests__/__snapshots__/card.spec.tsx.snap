// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders snapshot of content 1`] = `
.emotion-0 {
  display: block;
  position: relative;
  padding: 16px;
  background-color: #FFFFFF;
  border: 1px solid rgba(5, 25, 45, 0.15);
  border-radius: 4px;
  -webkit-transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),-webkit-transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
}

.emotion-0:where(a, button) {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0:hover {
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  -webkit-transform: translateY(-1px);
  -moz-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.emotion-0:focus-within {
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  -webkit-transform: translateY(-1px);
  -moz-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

<section
  class="emotion-0"
>
  <h3>
    Taylor Swift
  </h3>
  <p>
    Better than Jimmy Page.
  </p>
</section>
`;

exports[`Card renders snapshot of focused card 1`] = `
.emotion-0 {
  display: block;
  position: relative;
  padding: 16px;
  background-color: #FFFFFF;
  border: 1px solid rgba(5, 25, 45, 0.15);
  border-radius: 4px;
  -webkit-transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),-webkit-transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 2px #009BD8;
}

.emotion-0:where(a, button) {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0:hover {
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  -webkit-transform: translateY(-1px);
  -moz-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
  box-shadow: 0 0 0 2px #009BD8,0px 3px 5px -1px rgba(5, 25, 45, 0.3);
}

<a
  class="emotion-0"
  href="http://taylorswift.com"
>
  Taylor Swift Website
</a>
`;

exports[`Card renders snapshot of headstones and content 1`] = `
.emotion-0 {
  display: block;
  position: relative;
  padding: 16px;
  padding-top: 28px;
  background-color: #FFFFFF;
  border: 1px solid rgba(5, 25, 45, 0.15);
  border-radius: 4px;
  -webkit-transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),-webkit-transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
}

.emotion-0:where(a, button) {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0:hover {
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  -webkit-transform: translateY(-1px);
  -moz-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.emotion-0:focus-within {
  box-shadow: 0px 3px 5px -1px rgba(5, 25, 45, 0.3);
  -webkit-transform: translateY(-1px);
  -moz-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.emotion-1 {
  position: absolute;
  z-index: 1;
  top: -20px;
  left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
}

<section
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <span>
      Headstone 1
    </span>
    <span>
      Headstone 2
    </span>
  </div>
  <p>
    Taylor Swift
  </p>
</section>
`;

exports[`Card renders snapshot of the card with hover disabled 1`] = `
.emotion-0 {
  display: block;
  position: relative;
  padding: 16px;
  background-color: #FFFFFF;
  border: 1px solid rgba(5, 25, 45, 0.15);
  border-radius: 4px;
  -webkit-transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),-webkit-transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  transition: box-shadow 600ms cubic-bezier(0.1, 0.8, 0.2, 1),transform 600ms cubic-bezier(0.1, 0.8, 0.2, 1);
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
}

.emotion-0:where(a, button) {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<section
  class="emotion-0"
>
  <h3>
    Taylor Swift
  </h3>
  <p>
    Better than Jimmy Page.
  </p>
</section>
`;
